eclipse/src/saros/editor/EditorPool.java

```diff
@@ -237,7 +249,30 @@ public void remove(final IEditorPart editorPart) {
 
     editorManager.disconnect(file);
 
-    final saros.filesystem.IFile wrappedFile = ResourceAdapterFactory.create(file);
+    final saros.filesystem.IFile wrappedFile = convert(file);
```

Comment:
I do not understand why specific Core logic must be implemented here. I also discovered that the hashset uses core IFile, The whole logic here is Eclipse related, i.e it operates on editor parts and Eclipse IFiles.

It not the duty of this class to check any sharing etc.

Reply:
I introduced the usage of Saros `IFile` as part of the `SPath` removal, but it was using `SPath` objects beforehand, so that didn't really change a whole lot.

I don't see a problem in using Saros objects here. We are only interested in the editors of shared files anyways.

Furthermore, converting between Eclipse and Saros resource implementations is not as easy as it used to be. With the reference point implementations, we have to check every shared reference point by comparing its base path against the resource path. This creates a huge overhead compared to just wrapping the Eclipse object. So I would actually prefer directly using Saros objects here and unwrapping them if necessary instead of doing it the other way around.

This is also the reason why this class is "checking for sharing". With the new implementation, we can only ever create meaningful Saros resource objects for shared resources. Otherwise, we have no way of determining which directory to use as the base reference point to use.

Reply:
The check if this editor is shared is already done in the EditorManager. Why do you not just change the signature of the add method to require an IFile object beside the IEditorPart ?

Reply:
Will do.

Reply:
Done. Also applied to `EditorPool.add(...)` and removed the conversion logic from `EditorPool`.